### PR TITLE
More verbose logging around FindAT and SaveMsalResponse

### PR DESCRIPTION
**Pull request by bgavrilMS**
No perf impact on FindAccessToken logic...

Before
|                         Method | TokenCacheSize |     Mean |   Error |  StdDev |
|------------------------------- |--------------- |---------:|--------:|--------:|
| AcquireTokenForClientTestAsync |         100000 | 104.9 ms | 2.06 ms | 2.37 ms |

After
|                         Method | TokenCacheSize |     Mean |   Error |  StdDev |
|------------------------------- |--------------- |---------:|--------:|--------:|
| AcquireTokenForClientTestAsync |         100000 | 104.4 ms | 2.05 ms | 2.52 ms |

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs [hunk 1]
```diff
@@ -181,5 +183,43 @@ internal static MsalTokenResponse CreateFromAndroidBrokerResponse(string jsonRes
...
+                withoutPii.AppendLine("==MsalTokenResponse==");
+
+                withPii.AppendLine($"Error: {Error} ErrorDescription: {ErrorDescription}");
+                withoutPii.AppendLine($"Error: {Error} ErrorDescription: {ErrorDescription}");
+                withPii.AppendLine($"Scopes: {Scope} ");
+                withoutPii.AppendLine($"Scopes: {Scope} ");
+                withPii.AppendLine($"ExpiresIn: {ExpiresIn} RefreshIn {RefreshIn}");
+                withoutPii.AppendLine($"ExpiresIn: {ExpiresIn} RefreshIn {RefreshIn}");
+
+                withoutPii.AppendLine(
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:204]: I don't think we should log any token info or client info, and definitely not RTs, in either pii or non-pii logs. Has the guidance on this changed? 

what problem are we trying to solve? which issue is this linked to?

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:204]: There is a linked issue associated with the PR, it is https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2353

I am no logging any tokens, except for ClientInfo which is not a token. I'm only logging true or false if the token is present or not.

Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:204]: got it. my mistake, sorry about the false alarm.

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:204]: No prob, I'm happy that someone is looking at this important aspect in detail. Would not want to be the root cause of some security incident :p

**Review by trwalke — APPROVED** [review]

**Pull request merged**